### PR TITLE
Fix LinearRegression tests for scikit-learn v0.24

### PR DESCRIPTION
**Pull request by Pahandrovich**

### CODE REVIEW

**Review by owerbat — APPROVED** [review]

**Comment by PetrovKP** [comment]
Can we extend the interfaces without adding a new file, copying everything? Any ideas?

File: daal4py/sklearn/linear_model/_linear_0_24.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        super(LinearRegression, self).__init__(
+            fit_intercept=fit_intercept, normalize=normalize,
+            copy_X=copy_X, n_jobs=n_jobs, positive=positive)
+
+    def fit(self, X, y, sample_weight=None):
+        return _fit_copy(self, X, y, sample_weight=sample_weight)
+
+    def predict(self, X):
+        return _predict_copy(self, X)
+
```
Comment (Pahandrovich) [inline daal4py/sklearn/linear_model/_linear_0_24.py:252]: @PetrovKP, We can check sklearn version in lin_reg_0-23 wrapper in __init__, for transmission positive parameter. After that check parameter in fit method:

`class LinearRegression(LinearRegression_original):
    __doc__ = LinearRegression_original.__doc__

    def __init__(self, fit_intercept=True, normalize=False, copy_X=True,
                 n_jobs=None, positive=False):
        if (LooseVersion(sklearn_version) >= LooseVersion("0.24")):
            super(LinearRegression, self).__init__(
                fit_intercept=fit_intercept, normalize=normalize,
                copy_X=copy_X, n_jobs=n_jobs, positive=positive)
        else:
            super(LinearRegression, self).__init__(
                fit_intercept=fit_intercept, normalize=normalize,
                copy_X=copy_X, n_jobs=n_jobs)


    def fit(self, X, y, sample_weight=None):
        if self.positive == True:
            logging.info("sklearn.linar_model.LinearRegression.fit: " + method_uses_sklearn)
            return super(LinearRegression, self).fit(X, y=y, sample_weight=sample_weight)
        return _fit_copy(self, X, y, sample_weight=sample_weight)

    def predict(self, X):
        return _predict_copy(self, X)`

Comment (PetrovKP) [reply daal4py/sklearn/linear_model/_linear_0_24.py:252]: I like this option more, since we bring less copy-paste.

**Review by PetrovKP — APPROVED** [review]

**Comment by Pahandrovich** [comment]
This approach fails other tests.
https://app.circleci.com/pipelines/github/IntelPython/daal4py/1124/workflows/35a98312-77c2-49a8-8d0a-be9ec153e6df/jobs/1542

**Comment by Pahandrovich** [comment]
I will add the file for the new version, but with code reuse
